src/components/MyAccount/EditPopUpWindow.tsx

```diff
@@ -0,0 +1,222 @@
+import { useStoreSelector } from "@/store/hooks";
+import { updateUserInfo, userData } from "@/store/reducer/userSlice";
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useToast,
+} from "@chakra-ui/react";
+import { AxiosResponse } from "axios";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { updateUser, validPassword } from "../Login/helpers/userRequests";
+import validatePwd from "../Login/helpers/validatePwd";
+import EditInputGroup from "./EditInputGroup";
+
+const EditPopUpWindow = ({
+  onClose,
+  isOpen,
+  type,
+}: {
+  onClose: () => void;
+  isOpen: boolean;
+  type: string;
+}) => {
+  const [newFirstName, setNewFirstName] = useState("");
+  const [newLastName, setNewLastName] = useState("");
+  const [oldPsw, setOldPsw] = useState("");
+  const [newPsw, setNewPsw] = useState("");
+  const [confirmPsw, setConfirmPsw] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [weakPasswordMsg, setWeakPasswordMsg] = useState("");
+  const [incorrectPswMsg, setIncorrectPswMsg] = useState("");
+  const currentUserId = useStoreSelector(userData).userId;
+  const toast = useToast();
+  const dispatch = useDispatch();
+
+  const isBlank = (): boolean => {
+    switch (type) {
+      case "Name edit form":
```

Comment:
The type should be enum or const like below
```
enum FormType {
  NameEdit,
  PasswordEdit,
  ...
}
```
and the usage of the enum like below
```
switch (type) {
   case FormType.NameEdit : {
     // do some thing
     // break; or return;
   }
   ...
}
```

Reply:
Fix that, Please check.

Reply:
Good.